### ISSUE TITLE
Normal value for shaping loss

### ISSUE THREAD

**Issue by milliema** [zxhuang1698/interpretability-by-parts#6]
May I ask what are the normal value ranges for shaping loss on the three datasets? I tried the method ion my own dataset, yet the value keeps at 1.1 (with coeff of 0.5) and does no decrease. 
And what's the influences of hyperparameters in shaping loss?

**Comment by zxhuang1698** [comment]
Hi @milliema , thanks for your interest. I haven't touched this for a while, but I remember it's normally below 0.05. If the loss plateaus, it might be the case that there are no enough easily-separable concepts for your dataset compared to the number of concepts you set. Also, to make the model better converge, you can try to load a pre-trained model (e.g. first three residual block of ResNet) instead of training from scratch.

Besides, you can play with coefficient, random seed, or [epsilon in the loss](https://github.com/zxhuang1698/interpretability-by-parts/blob/650f1af573075a41f04f2f715f2b2d4bc0363d31/src/common/loss.py#L61) to see whether they lead to better convergence on your dataset.

**Comment by milliema** [comment]
Thanks for your prompt response, I'm working on the face recognition dataset actually, and trying to find out if the region grouping idea works for face identity classification. The num_parts is 9, following the settings in Celeb dataset. I'll try the suggested solutions you've mentioned, thanks!

**Comment by snant** [comment]
how to alter the input image in visualize，input image is 25 by defult in your code

**Comment by zxhuang1698** [comment]
You can try to change the number of columns and rows in the [visualization code](https://github.com/zxhuang1698/interpretability-by-parts/blob/650f1af573075a41f04f2f715f2b2d4bc0363d31/src/cub200/visualize.py#L167).
